### PR TITLE
fix(codewhisperer): inlay not supported error in injected editior

### PR DESCRIPTION
**Pull request by zixlin7**
before:

https://github.com/aws/aws-toolkit-jetbrains/assets/60411978/0a2097e9-e3f3-48af-97bc-e61eba475567

after:

https://github.com/aws/aws-toolkit-jetbrains/assets/60411978/dd6681d3-06d5-4127-9299-cacabb6f8c5c

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Checklist
- [ ] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [ ] A short description of the change has been added to the **[CHANGELOG](https://github.com/aws/aws-toolkit-jetbrains/blob/master/CONTRIBUTING.md#contributing-via-pull-requests)** if the change is customer-facing in the IDE.
- [ ] I have added metrics for my changes (if required)
 
## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: jetbrains-core/src/software/aws/toolkits/jetbrains/utils/PsiUtils.kt [hunk 1]
```diff
@@ -10,11 +10,16 @@ import com.intellij.psi.PsiElement
 fun PsiElement.isTestOrInjectedText(): Boolean {
     val project = this.project
     val virtualFile = this.containingFile.virtualFile ?: return false
-    if (virtualFile is VirtualFileWindow) {
+    if (this.isInjectedText() || ProjectRootManager.getInstance(project).fileIndex.isInTestSourceContent(virtualFile)) {
```
Comment (andrewyuq) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/utils/PsiUtils.kt:13]: `ProjectRootManager.getInstance(project).fileIndex.isInTestSourceContent(virtualFile)`
what does this mean?

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/service/CodeWhispererService.kt [hunk 2]
```diff
@@ -93,6 +94,8 @@ class CodeWhispererService {
             }
             return
         }
+        val isInjectedFile = runReadAction { psiFile.isInjectedText() }
+        if (isInjectedFile) return
```
Comment (andrewyuq) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/service/CodeWhispererService.kt:98]: do we want to show anything on manual trigger?

File: jetbrains-core/src/software/aws/toolkits/jetbrains/utils/PsiUtils.kt (hunk 1 above)
Comment (zixlin7) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/utils/PsiUtils.kt:13]: I'm not sure what's the use case for this, I didn't touch this part from the original function, I just extracted the injected check into a separate function so we can use

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/service/CodeWhispererService.kt (hunk 2 above)
Comment (zixlin7) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/service/CodeWhispererService.kt:98]: this bug doesn't happen for manual trigger, not sure the reason but manual trigger always work

File: jetbrains-core/src/software/aws/toolkits/jetbrains/utils/PsiUtils.kt (hunk 1 above)
Comment (andrewyuq) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/utils/PsiUtils.kt:13]: ah I see below

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/service/CodeWhispererService.kt (hunk 2 above)
Comment (Will-ShaoHua) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/service/CodeWhispererService.kt:98]: interesting case, so manual trigger is able to display inlay where auto doesn't

Comment (andrewyuq) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/service/CodeWhispererService.kt:98]: it must be that at the entry points the editor for auto trigger is a different one than the editor for manual trigger

**Review by andrewyuq — APPROVED** [review]

**Review by Will-ShaoHua — APPROVED** [review]

**Review by manodnyab — APPROVED** [review]

**Pull request merged**